taskfile/cache.go

```diff
@@ -2,24 +2,51 @@ package taskfile
 
 import (
 	"crypto/sha256"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 )
 
+// DefaultCacheTTL is used when a value is not explicitly given to a new Cache.
+const DefaultCacheTTL = time.Duration(time.Hour * 24)
+
 type Cache struct {
 	dir string
+	ttl time.Duration
 }
 
-func NewCache(dir string) (*Cache, error) {
+// ErrExpired is returned when a cached file has expired.
+var ErrExpired = errors.New("task: cache expired")
```

Comment:
It should go in the `errors` package.
You need to be a bit more specific in the error message, for example write wich file is expired